src/Input/InputWithIcon.module.scss

```diff
@@ -1,36 +1,53 @@
 @import '../variables.scss';
 
+.icon {
+  box-sizing: content-box;
+  width: 14px;
+  position: absolute;
+  top: 0;
+  padding: 0 10px;
+  height: 100%;
+  fill: var(--casper);
+
+  &.left {
+    left: 0;
+  }
+
+  &.right {
+    right: 0;
+  }
+}
+
 .wrapper {
-    position: relative;
+  position: relative;
 
-    &:hover .icon {
-        fill: var(--green);
+  &:hover .icon {
+    fill: var(--green);
+  }
+
+  &.disabled {
+    & .icon {
+      fill: var(--mystic);
```

Comment:
If we use `[disabled] + .icon`, then `disabled` property destruction could be omitted.